server/src/test/java/server/haengdong/presentation/EventControllerTest.java

```diff
@@ -46,6 +55,32 @@ void saveEvent() throws Exception {
                         .content(requestBody))
                 .andDo(print())
                 .andExpect(status().isOk())
-                .andExpect(MockMvcResultMatchers.redirectedUrl("events/" + token));
+                .andExpect(redirectedUrl("events/" + token));
+    }
+
+    @DisplayName("토큰으로 행사의 모든 액션을 조회한다.")
+    @Test
+    void findActionsTest() throws Exception {
+        String token = "TOKEN";
+        Event event = new Event("행동대장 회식", token);
+        Action action1 = new Action(event, 1L);
+        MemberAction memberAction = new MemberAction(action1, "소하", MemberActionStatus.IN, 1L);
+        Action action2 = new Action(event, 2L);
+        BillAction billAction = new BillAction(action2, "뽕나무쟁이족발", 30000L);
+        given(eventService.findActions(token)).willReturn(
+                List.of(ActionAppResponse.of(memberAction), ActionAppResponse.of(billAction)));
+
+        mockMvc.perform(get("/api/events/" + token + "/actions"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.actions").isArray())
+                .andExpect(jsonPath("$.actions[0].name").value("소하"))
+                .andExpect(jsonPath("$.actions[0].price").isEmpty())
+                .andExpect(jsonPath("$.actions[0].sequence").value(1L))
+                .andExpect(jsonPath("$.actions[0].type").value("IN"))
+                .andExpect(jsonPath("$.actions[1].name").value("뽕나무쟁이족발"))
+                .andExpect(jsonPath("$.actions[1].price").value(30000L))
+                .andExpect(jsonPath("$.actions[1].sequence").value(2L))
+                .andExpect(jsonPath("$.actions[1].type").value("BILL"));
```

Comment:
```suggestion
        List<ActionAppResponse> actionAppResponses = List.of(ActionAppResponse.of(memberAction),
                ActionAppResponse.of(billAction));
        given(eventService.findActions(token)).willReturn(actionAppResponses);

        ActionsResponse expectedActions = ActionsResponse.of(actionAppResponses);
        String expectedJson = objectMapper.writeValueAsString(expectedActions);

        mockMvc.perform(get("/api/events/" + token + "/actions"))
                .andDo(print())
                .andExpect(status().isOk())
                .andExpect(content().json(expectedJson));
```
c: 이렇게 수정하면 다른 테스트랑 컨벤션이 다르긴 한데 ㅎㅎ... 여기 andExpect가 너무 많아요 알잘딱깔쎈 해 주세요

Reply:
감자가 어떤 의미로 코멘트 달았는지 이해하지만, 테스트 검증부를 개발자가 못보는 것이 어색해서 현재 코드 유지하려고 합니다. 👍